catroid/src/main/java/org/catrobat/catroid/utils/CrashReporter.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.utils;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.support.annotation.VisibleForTesting;
+import android.util.Log;
+
+import com.crashlytics.android.Crashlytics;
+import com.google.gson.Gson;
+
+import org.catrobat.catroid.BuildConfig;
+import org.catrobat.catroid.ui.MainMenuActivity;
+import org.catrobat.catroid.ui.SettingsActivity;
+
+import io.fabric.sdk.android.Fabric;
+
+public class CrashReporter {
+
+	Context context;
+	private SharedPreferences preferences;
+	public boolean isAnonymousReportEnabled;
+	public boolean isCrashReportEnabled;
+	public boolean isRecoveredFromCrash;
+
+	public static final String RECOVERED_FROM_CRASH = "RECOVERED_FROM_CRASH";
+	public static final String EXCEPTION_FOR_REPORT = "EXCEPTION_FOR_REPORT";
+
+	private static final String TAG = CrashReporter.class.getSimpleName();
+
+	public CrashReporter(Context context) {
+		this.context = context;
+		preferences = PreferenceManager.getDefaultSharedPreferences(context);
+		setIsAnonymousCrashReportEnable(preferences.getBoolean(SettingsActivity.SETTINGS_CRASH_REPORTS, false));
+		setIsCrashReportEnabled(BuildConfig.CRASHLYTICS_CRASH_REPORT_ENABLED);
+		setIsRecoveredFromCrash(preferences.getBoolean(RECOVERED_FROM_CRASH, false));
+	}
+
+	public boolean initialize() {
+		if (isAnonymousReportEnabled && isCrashReportEnabled) {
+			Fabric.with(context, new Crashlytics());
+			Log.d(TAG, "INITIALIZED!");
+			return true;
+		}
+		Log.d(TAG, "INITIALIZATION FAILED!");
+		return false;
+	}
+
+	@VisibleForTesting
+	public void setIsAnonymousCrashReportEnable(boolean value) {
```

Comment:
Please remove. It's not used for testing anymore. The whole method can be inlined.